bundles/org.openhab.binding.emotiva/src/main/java/org/openhab/binding/emotiva/internal/EmotivaProcessorHandler.java

```diff
@@ -0,0 +1,787 @@
+                    && stateMap.get(EmotivaSubscriptionTags.keepAlive.name()) instanceof Number keepAlive
+                            ? keepAlive.intValue()
+                            : config.keepAlive;
+            pollingJob = scheduler.scheduleWithFixedDelay(this::checkKeepAliveTimestamp,
+                    delay + DEFAULT_KEEP_ALIVE_IN_MILLISECONDS, delay + DEFAULT_KEEP_ALIVE_IN_MILLISECONDS,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void checkKeepAliveTimestamp() {
+        if (ThingStatus.ONLINE.equals(getThing().getStatusInfo().getStatus())) {
+            State state = stateMap.get(LAST_SEEN_STATE_NAME);
+            if (state instanceof Number value) {
+                Instant lastKeepAliveMessageTimestamp = Instant.ofEpochSecond(value.longValue());
+                Instant deviceGoneGracePeriod = Instant.now().minus(config.keepAlive, ChronoUnit.MILLIS)
+                        .minus(DEFAULT_KEEP_ALIVE_CONSIDERED_LOST_IN_MILLISECONDS, ChronoUnit.MILLIS);
+                if (lastKeepAliveMessageTimestamp.isBefore(deviceGoneGracePeriod)) {
+                    logger.debug(
+                            "Last KeepAlive message received '{}', over grace-period by '{}', consider '{}' gone, setting OFFLINE and disposing",
+                            lastKeepAliveMessageTimestamp,
+                            Duration.between(lastKeepAliveMessageTimestamp, deviceGoneGracePeriod),
+                            thing.getThingTypeUID());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/message.processor.connection.error.keep-alive");
+                    // Connection lost, avoid sending unsubscription messages
+                    udpSenderActive = false;
+                    disconnect();
+                    scheduleConnectRetry(retryConnectInMinutes);
+                }
+            }
+        } else if (ThingStatus.OFFLINE.equals(getThing().getStatusInfo().getStatus())) {
+            logger.debug("Keep alive pool job, '{}' is '{}'", getThing().getThingTypeUID(),
+                    getThing().getStatusInfo().getStatus());
+        }
+    }
+
+    private void handleStatusUpdate(EmotivaUdpResponse emotivaUdpResponse) {
+        udpSenderActive = true;
+        logger.debug("Received data from '{}' with length '{}'", emotivaUdpResponse.ipAddress(),
+                emotivaUdpResponse.answer().length());
+
+        Object object;
+        try {
+            object = xmlUtils.unmarshallToEmotivaDTO(emotivaUdpResponse.answer());
+        } catch (JAXBException e) {
+            logger.debug("Could not unmarshal answer from '{}' with length '{}' and content '{}'",
+                    emotivaUdpResponse.ipAddress(), emotivaUdpResponse.answer().length(), emotivaUdpResponse.answer(),
+                    e);
+            return;
+        }
+
+        if (object instanceof EmotivaAckDTO answerDto) {
```

Comment:
```suggestion
        if (object instanceof EmotivaAckDTO) {
```

either remove the unused var or (little bit more work but better), override the `EmotivaAckDTO.toString()` and replace `emotivaUdpResponse.answer()` with `answerDto`

Reply:
Added an override on the EmotivaAckDTO.toString()